reference/docs-conceptual/dev-cross-plat/vscode/using-vscode-for-debugging-compiled-cmdlets.md

```diff
@@ -0,0 +1,204 @@
+---
+ms.date:  10/19/2020
+keywords:  powershell,cmdlet,debug
+title:  Using Visual Studio Code for debugging compiled cmdlets
+description: How to set a build task and launch configuration for a PSModule project in .NET Core
+---
+# Using Visual Studio Code for debugging compiled cmdlets
+
+This guide shows you how to interactively debug C# source code for a compiled PowerShell module
+using Visual Studio Code and the C# extension.
+
+Some familiarity with the Visual Studio Code debugger is assumed.
+
+- For a general introduction to the Visual Studio Code debugger, see
+  [Debugging in Visual Studio Code][].
+
+- For examples of debugging PowerShell script files and modules, see
+  [Using Visual Studio Code for remote editing and debugging][].
+
+This guide follows on from the [Writing Portable Modules][] guide.
+
+## Creating a build task
+
+Build your project automatically before launching a debugging session. Rebuilding ensures that you
+debug the latest version of your code.
+
+Configure a build task:
+
+1. In the **Command Palette**, run the **Configure Default Build Task** command.
+
+   ![Run Configure Default Build Task](media/using-vscode-for-debugging-compiled-cmdlets/configure-default-build-task.png)
+
+1. In the **Select a task to configure** dialog, choose **Create tasks.json file from template**.
+
+1. In the **Select a Task Template** dialog, choose **.NET Core**.
+
+A new `tasks.json` file is created if one doesn't exist yet.
+
+To test your build task:
+
+1. In the **Command Palette**, run the **Run Build Task** command.
+
+1. In the **Select the build task to run** dialog, choose **build**.
+
+### Information about DLL files being locked
+
+By default, a successful build doesn't show output in the terminal pane. If you see output that
+contains the text **Project file doesn't exist**, you should edit the `tasks.json` file. Include the
+explicit path to the C# project expressed as `"${workspaceFolder}/myModule"`. In this example,
+`myModule` is the name of the project folder. This entry must go after the `build` entry in the
+`args` list as follows:
+
+```json
+    {
+        "label": "build",
+        "command": "dotnet",
+        "type": "shell",
+        "args": [
+            "build",
+            "${workspaceFolder}/myModule",
+            // Ask dotnet build to generate full paths for file names.
+            "/property:GenerateFullPaths=true",
+            // Do not generate summary otherwise it leads to duplicate errors in Problems panel
+            "/consoleloggerparameters:NoSummary",
+        ],
+        "group": "build",
+        "presentation": {
+            "reveal": "silent"
+        },
+        "problemMatcher": "$msCompile"
+    }
+```
+
+When debugging, your module DLL is imported into the PowerShell session in the VS Code terminal. The
```

Comment:
Let's spell out VS Code as Visual Studio Code to maintain consistency throughout the article.

Reply:
Yes I think the vscode marketing team prefers this (though this info is from 2 Ignites ago so may not be the case anymore)

Reply:
"VS Code" (not "VSCode") is allowable but only after using the full name. Fixed